server/src/storage/datastore.ts

```diff
@@ -55,4 +56,37 @@ const getAllWithId = async (kind: string) => {
   return resWithId;
 };
 
-export {getWithId, getAllWithId};
+/**
+ * Unpacks id of modified object from MutationResult object.
+ */
+const getIdFromMutationResult = (
+  mutationResult: google.datastore.v1.IMutationResult
+): number | Long | string | null | undefined =>
```

Comment:
why this is return type? same type of id in mutationResult?

Reply:
Yep, the Datastore id has return type `number | Long | string | null`. I refactored this out into a `ResponseId` type.